From a65ee4c5e4c43f7b50695da9f6296077963772e4 Mon Sep 17 00:00:00 2001
From: Emil Medve <Emilian.Medve@Freescale.com>
Date: Fri, 17 Jan 2014 10:35:39 -0600
Subject: [PATCH 512/987] fsl_qbman: Check kmalloc() result

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

This fixes the following coccinelle warning:

alloc with no test

Signed-off-by: Emil Medve <Emilian.Medve@Freescale.com>
Change-Id: I573f5fb75cc8b88336bf96e18464730dd8c79686
Reviewed-on: http://git.am.freescale.net:8181/8107
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Haiying Wang <Haiying.Wang@freescale.com>
Reviewed-by: Emilian Medve <Emilian.Medve@freescale.com>
---
 drivers/staging/fsl_qbman/bman_high.c | 2 ++
 drivers/staging/fsl_qbman/qman_high.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/staging/fsl_qbman/bman_high.c b/drivers/staging/fsl_qbman/bman_high.c
index bcfc5e8..16e9534 100644
--- a/drivers/staging/fsl_qbman/bman_high.c
+++ b/drivers/staging/fsl_qbman/bman_high.c
@@ -205,6 +205,8 @@ struct bman_portal *bman_create_portal(
 
 	if (!portal) {
 		portal = kmalloc(sizeof(*portal), GFP_KERNEL);
+		if (!portal)
+			return portal;
 		portal->alloced = 1;
 	} else
 		portal->alloced = 0;
diff --git a/drivers/staging/fsl_qbman/qman_high.c b/drivers/staging/fsl_qbman/qman_high.c
index d7f2a9c..0c4dac2 100644
--- a/drivers/staging/fsl_qbman/qman_high.c
+++ b/drivers/staging/fsl_qbman/qman_high.c
@@ -373,6 +373,8 @@ struct qman_portal *qman_create_portal(
 
 	if (!portal) {
 		portal = kmalloc(sizeof(*portal), GFP_KERNEL);
+		if (!portal)
+			return portal;
 		portal->alloced = 1;
 	} else
 		portal->alloced = 0;
-- 
1.9.1

